### PR TITLE
feat(linter): add a check for affected field

### PR DESCRIPTION
**Pull request by hogo6002**
`affected:` should be defined in the data source and should not be an empty array.

### CODE REVIEW

File: tools/osv-linter/internal/checks/affected.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package checks
+
+import (
+	"github.com/tidwall/gjson"
+)
+
+var CheckAffectedFieldValid = &CheckDef{
+	Code:        "A0001",
+	Name:        "affected-field-valid",
+	Description: "affected field validates",
```
Comment (andrewpollock) [inline tools/osv-linter/internal/checks/affected.go:10]: The style I've been going with is more like a test assertion

```suggestion
	Description: "every record has affected data",
```

File: tools/osv-linter/internal/checks/affected.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package checks
+
+import (
+	"github.com/tidwall/gjson"
+)
+
+var CheckAffectedFieldValid = &CheckDef{
+	Code:        "A0001",
+	Name:        "affected-field-valid",
```
Comment (andrewpollock) [inline tools/osv-linter/internal/checks/affected.go:9]: This names it more like the existing `introduced-event-exists` check"

```suggestion
	Name:        "affected-data-exists",
```

File: tools/osv-linter/internal/checks/affected.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	Description: "affected field validates",
+	Check:       AffectedFieldValid,
```
Comment (andrewpollock) [inline tools/osv-linter/internal/checks/affected.go:11]: Naming nit.

```suggestion
	Check:       RecordHasAffected,
```

File: tools/osv-linter/internal/checks/affected.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+// AffectedFieldValid checks if the 'affected' field exists in the JSON and is not an empty array.
+func AffectedFieldValid(json *gjson.Result, config *Config) (findings []CheckError) {
```
Comment (andrewpollock) [inline tools/osv-linter/internal/checks/affected.go:15]: Naming nit.

```suggestion
// RecordHasAffected checks if the 'affected' field exists in the JSON and is not an empty array.
func RecordHasAffected(json *gjson.Result, config *Config) (findings []CheckError) {
```

File: tools/osv-linter/internal/checks/affected.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package checks
+
+import (
+	"github.com/tidwall/gjson"
+)
+
+var CheckAffectedFieldValid = &CheckDef{
+	Code:        "A0001",
```
Comment (andrewpollock) [inline tools/osv-linter/internal/checks/affected.go:8]: (Over) thinking about codes, and prefixes, IMO, this is a record-level check, but I'm already using "R" for ranges...

**Review by andrewpollock — COMMENTED** [review]
This LGTM as is, all of my feedback is on naming.

File: tools/osv-linter/internal/checks/affected.go (hunk 5 above)
Comment (hogo6002) [reply tools/osv-linter/internal/checks/affected.go:8]: I also feel I shouldn't name this file as `affected.go`?. `record.go` sounds better and can add more checks in the future.

Comment (hogo6002) [reply tools/osv-linter/internal/checks/affected.go]: I renamed the file to `record.go` and renamed the check to `CheckRecordHasAffected`, but i didn't change the code. I agree that this is a record-level check and "R" is better than "A". Should we rename the range code to "V" for versions?

Comment (andrewpollock) [reply tools/osv-linter/internal/checks/affected.go]: > Should we rename the range code to "V" for versions?

Versions are one of the types of things in ranges :-)

Maybe for record-level, we go with the very non-obvious "O" for OSV? Naming is hard...

**Review by andrewpollock — APPROVED** [review]
LGTM, this is good enough for now. We can worry about improving the code later (famous last words)

**Pull request merged**